core/src/transport/choice.rs

```diff
@@ -51,13 +52,37 @@ where
         id: ListenerId,
         addr: Multiaddr,
     ) -> Result<(), TransportError<Self::Error>> {
+        trace!(
+            "Attempting to dial {} using {}",
+            addr,
+            std::any::type_name::<A>()
+        );
         let addr = match self.0.listen_on(id, addr) {
-            Err(TransportError::MultiaddrNotSupported(addr)) => addr,
+            Err(TransportError::MultiaddrNotSupported(addr)) => {
+                debug!(
+                    "Failing to dial {} using {}",
```

Comment:
```suggestion
                    "Failed to dial {} using {}",
```

Nit: at this point it already failed, right?